src/Service.Tests/Configuration/HotReload/ConfigurationHotReloadTests.cs

```diff
@@ -299,4 +302,127 @@ public async Task HotReloadConfigRuntimeGQLEnabledEndToEndTest()
+    }
+
+    /// <summary>
+    /// Hot reload the configuration file so that it changes from one connection string
+    /// to an invalid connection string, then it hot reloads once more to the original
+    /// connection string. Lastly, we assert that the first reload fails while the second one succeeds.
+    /// </summary>
+    [TestCategory(MSSQL_ENVIRONMENT)]
+    [TestMethod]
+    public async Task HotReloadConfigConnectionStringEndToEndTest()
+    {
+        // Arrange
+        _writer = new StringWriter();
+        Console.SetOut(_writer);
+
+        string failedKeyWord = "Unable to hot reload configuration file due to";
+        string succeedKeyWord = "Validated hot-reloaded configuration file";
+
+        // Act
+        // Hot Reload should fail here
+        GenerateConfigFile(
+            connectionString: "");
+        System.Threading.Thread.Sleep(5000);
+
+        // Log that shows that hot-reload was not able to validate properly
+        string failedConfigLog = $"{_writer.ToString()}";
+
+        // Hot Reload should succeed here
+        GenerateConfigFile(
+            connectionString: $"{ConfigurationTests.GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL).Replace("\\", "\\\\")}");
+        System.Threading.Thread.Sleep(8000);
+
+        // Log that shows that hot-reload validated properly
+        string succeedConfigLog = $"{_writer.ToString()}";
+
+        HttpResponseMessage restResult = await _testClient.GetAsync("/rest/Book");
+
+        // Assert
+        Assert.IsTrue(failedConfigLog.Contains(failedKeyWord));
+        Assert.IsTrue(succeedConfigLog.Contains(succeedKeyWord));
+        Assert.AreEqual(HttpStatusCode.OK, restResult.StatusCode);
+    }
+
+    /// <summary>
+    /// /// (Warning: This test only currently works in the pipeline due to constrains of not
+    /// being able to change from one database type to another, under normal circumstances
+    /// hot reload allows changes from one database type to another)
+    /// Hot reload the configuration file so that it changes from one database type to another.
+    /// Then it hot reloads once more to the original database type. We assert that the
+    /// first reload fails while the second one succeeds.
+    /// </summary>
+    [TestCategory(MSSQL_ENVIRONMENT)]
+    [TestMethod]
+    public async Task HotReloadConfigDatabaseTypeEndToEndTest()
+    {
+        // Arrange
+        _writer = new StringWriter();
+        Console.SetOut(_writer);
+
+        string failedKeyWord = "Unable to hot reload configuration file due to";
+        string succeedKeyWord = "Validated hot-reloaded configuration file";
+
+        // Act
+        // Hot Reload should fail here
+        GenerateConfigFile(
+            databaseType: DatabaseType.PostgreSQL,
+            connectionString: $"{ConfigurationTests.GetConnectionStringFromEnvironmentConfig(TestCategory.POSTGRESQL).Replace("\\", "\\\\")}");
+        System.Threading.Thread.Sleep(5000);
```

Comment:
why someplace sleep time is 5000, someplace its 8000? was the test failing?

Reply:
Yes, the test was failing and it was due to a lack of enough time for it to complete

Reply:
Changed the order of some operations and changed sleep from 8000 to 5000 to make test less flaky